scio-test/src/test/scala/com/spotify/scio/values/PairHashSCollectionFunctionsTest.scala

```diff
@@ -77,9 +77,22 @@ class PairHashSCollectionFunctionsTest extends PipelineSpec {
     runWithContext { sc =>
       val p1 = sc.parallelize(Seq(("a", 1), ("a", 2), ("b", 3)))
       val p2 = sc.parallelize[(String, Int)](Map.empty).asMultiMapSideInput
+      val p = p1.hashJoin(p2)
+      p should containInAnyOrder(Seq.empty[(String, (Int, Int))])
+    }
+  }
+
+  it should "support hashJoin() with transformed .asMultiMapSideInput" in {
+    runWithContext { sc =>
+      val p1 = sc.parallelize(Seq(("a", 1), ("a", 2), ("b", 3)))
+      val p2 = sc
+        .parallelize(Seq(("a", "11"), ("b", "12"), ("b", "13")))
+        .asMultiMapSideInput
+        .map(_.map { case (k, vs) => k -> vs.map(_.toInt) })
```

Comment:
This transformation was failing, because we tried to use a `Coder[String]` to encode the produced `Int` value